main.go

```diff
@@ -358,10 +358,11 @@ func main() {
 	}
 	subcommands = append(subcommands, cmdUpdateResolvers)
 
-	// add --debug and --timings flag to every command
+	// add --debug, --timings and --experiment flag to every command
 	for _, cmd := range subcommands {
 		cmd.Flags().BoolVarP(&burrito.PrintStackTrace, "debug", "", false, "Enables debugging")
 		cmd.Flags().BoolVarP(&regolith.EnableTimings, "timings", "", false, "Enables timing information")
+		cmd.Flags().StringSliceVar(&regolith.EnabledExperiments, "experiments", nil, "Enables experimental features")
```

Comment:
We should provide information about the available experiments in help.

Reply:
I added additional info for the `experiments` flag in commit 03dd528cee2e8c0ff15e3543aa7a809854d847e8. This is how the help message looks now:
![image](https://github.com/Bedrock-OSS/regolith/assets/34062025/fad37289-d453-4806-8b1c-56c931f14a38)

I'm not happy about this "experiments" flag idea. I think we could just enable the optimization from this PR by default and add a flag to disable it. It's not like we have a lot of experimental features being available all the time.

Reply:
I'm in favor of making this optimization default + adding an option to disable it, but I want to keep experiments flag for the future. I do want to have more room to experiment with different ideas. This PR can keep the flag and support for it, but also make this optimization as a default thing.

Reply:
Let's keep the flag for the next release and change it to the default behavior in the future. The first run with 'size_time_check' is probably slower than the first run without it so making it default will affect the performance of CI with github actions that use Regolith. I need to remember that in some of my projects...